Update Vivotek component stream source

HarlemSquirrel opened the pull request:
## Description:

- Fix building stream URL
- Add new `stream_path` configuration option

**Related issue (if applicable):** fixes #26942

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#10899

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [x] Untested files have been added to `.coveragerc`.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

HarlemSquirrel committed:
Update Vivotek component

Fix building stream URL

HarlemSquirrel committed:
Update Vivotek component

Make stream path optionally configurable.

cgtobi commented on homeassistant/components/vivotek/camera.py:
```diff
@@ -49,10 +53,13 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
+                config[CONF_PASSWORD],
+                config[CONF_IP_ADDRESS],
+                config[CONF_STREAM_PATH],
+            )
```
Why not actually use f-strings here?

HarlemSquirrel replied:
I am pretty new to Python so I was not aware of f-strings. I just read through https://realpython.com/python-f-strings/ and they look pretty cool! The article did say they were introduced in Python 3.6 so as long as we don't need this code to work in < 3.6 I am happy to make the change.

cgtobi replied:
We dropped support for python <3.6, so f-strings are fine.

HarlemSquirrel replied:
Updated

cgtobi reviewed (approved):
LGTM

HarlemSquirrel committed:
Update Vivotek camera integration

Use f-string to build stream source URL.
This improve readability and I hear it runs faster.

cgtobi reviewed (approved):
LGTM

MartinHjelmare commented on homeassistant/components/vivotek/camera.py:
```diff
@@ -33,12 +36,14 @@
         vol.Optional(CONF_SSL, default=False): cv.boolean,
         vol.Optional(CONF_VERIFY_SSL, default=True): cv.boolean,
         vol.Optional(CONF_FRAMERATE, default=2): cv.positive_int,
+        vol.Optional(CONF_STREAM_PATH, default=DEFAULT_STREAM_SOURCE): cv.string,
```
Is it impossible to detect this automatically?

HarlemSquirrel replied:
This allows choosing a different stream path mostly to allow using a different encoding. I don't see a clear standard for this at the moment so that's why it's open-ended here.

MartinHjelmare replied:
Ok.
